mlos_bench/mlos_bench/optimizers/mlos_core_optimizer.py

```diff
@@ -99,26 +95,41 @@ def bulk_register(self,
                       configs: Sequence[dict],
                       scores: Sequence[Optional[Dict[str, TunableValue]]],
                       status: Optional[Sequence[Status]] = None) -> bool:
+
         if not super().bulk_register(configs, scores, status):
             return False
+
         df_configs = self._to_df(configs)  # Impute missing values, if necessary
-        df_scores = pd.Series(
-            [self._extract_target(score) for score in scores],
-            dtype=float) * self._opt_sign
+
+        df_scores = self._adjust_signs_df(
+            pd.DataFrame([{} if score is None else score for score in scores]))
+
+        opt_targets = list(self._opt_targets)
         if status is not None:
+            # Select only the completed trials, set scores for failed trials to +inf.
             df_status = pd.Series(status)
-            df_scores[df_status != Status.SUCCEEDED] = float("inf")
+            df_scores.loc[df_status != Status.SUCCEEDED, opt_targets] = float("inf")
```

Comment:
We may still want to change this to something less awful, per #523.

For instance, 3-10x the worst *actually evaluated* sample recorded so far was shown in at least one paper I remember as performing better for models to learn on.

Reply:
Maybe just add a comment to that effect for now.

Reply:
Or could get handled in the base class somehow.
In any case, more a note for a future PR than anything else.

Reply:
agree. I've added comments with the link to #523 in the places where we replace values with +inf